server/src/main/java/io/druid/segment/realtime/appenderator/AppenderatorDriver.java

```diff
@@ -682,6 +356,59 @@ public void onFailure(Throwable t)
             );
           }
 
+          return segmentsAndMetadata;
+        },
+        executor
+    );
+  }
+
+  /**
+   * Drop segments in background. The segments should be pushed (in batch ingestion) or published (in streaming
+   * ingestion) before being dropped.
+   *
+   * @param segmentsAndMetadata result of pushing or publishing
+   *
+   * @return a future for dropping segments
+   */
+  ListenableFuture<SegmentsAndMetadata> dropInBackground(SegmentsAndMetadata segmentsAndMetadata)
+  {
+    log.info("dropping segments[%s]", segmentsAndMetadata.getSegments());
```

Comment:
Please format the log message a bit nicer (capitalization)

Reply:
Done.